test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go

```diff
@@ -52,30 +76,436 @@ var _ = Describe("[performance] Cgroups and affinity", Ordered, func() {
+				// and check if activation files is modified or deleted
+				profile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+				Expect(err).ToNot(HaveOccurred(), "Unable to fetch latest performance profile")
+				currentPolicy := profile.Spec.NUMA.TopologyPolicy
+				if *currentPolicy == "best-effort" {
+					policy = "restricted"
+				}
+				profile.Spec.NUMA = &performancev2.NUMA{
+					TopologyPolicy: &policy,
+				}
+				By("Updating the performance profile")
+				profiles.UpdateWithRetry(profile)
+				By("Applying changes in performance profile and waiting until mcp will start updating")
+				mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+				By("Waiting for MCP being updated")
+				mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+				By("Checking Activation file")
+				cmd := []string{"ls", activation_file}
+				for _, node := range workerRTNodes {
+					out, err := nodes.ExecCommandOnNode(cmd, &node)
+					Expect(err).ToNot(HaveOccurred(), "file %s doesn't exist ", activation_file)
+					Expect(out).To(Equal(activation_file))
+				}
+			})
+			AfterEach(func() {
+				By("Reverting the Profile")
+				profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+				Expect(err).ToNot(HaveOccurred())
+				currentSpec, _ := json.Marshal(profile.Spec)
+				spec, _ := json.Marshal(initialProfile.Spec)
+				performanceMCP, err := mcps.GetByProfile(profile)
+				Expect(err).ToNot(HaveOccurred())
+				// revert only if the profile changes.
+				if !bytes.Equal(currentSpec, spec) {
+					Expect(testclient.Client.Patch(context.TODO(), profile,
+						client.RawPatch(
+							types.JSONPatchType,
+							[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec", "value": %s }]`, spec)),
+						),
+					)).ToNot(HaveOccurred())
+
+					By("Applying changes in performance profile and waiting until mcp will start updating")
+					mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
 
-		It("Verify OVS slice is configured correctly", func() {
-			By("checking the cpuset spans all cpus")
+					By("Waiting when mcp finishes updates")
+					mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+				}
+			})
+		})
+	})
+
+	Context("Verification of cgroup layout on the worker node", func() {
+		chkOvsCgrpProcs := func(node *corev1.Node) (string, error) {
+			testlog.Info("Verify cgroup.procs is not empty")
+			cmd := []string{"cat", "/rootfs/sys/fs/cgroup/cpuset/ovs.slice/cgroup.procs"}
+			return nodes.ExecCommandOnNode(cmd, node)
+		}
+		chkOvsCgrpCpuset := func(node *corev1.Node) (string, error) {
 			cmd := []string{"cat", "/rootfs/sys/fs/cgroup/cpuset/ovs.slice/cpuset.cpus"}
-			ovsCpuList, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
+			return nodes.ExecCommandOnNode(cmd, workerRTNode)
```

Comment:
The `chkOvsCgrpCpuset` function's  parameter name is `node` but you are passing `workerRTNode` to the 
 `nodes.ExecCommandOnNode` function instead.